### ISSUE TITLE
Getting different recommendations using "Predict" and "dot product of user and item"

### ISSUE THREAD

**Issue by kshitijyad** [lyst/lightfm#474]
Thanks for opening an issue!

Please include as much detail as possible: what does your dataset look like, what hyperparameters you are using (and have you tried other ones?).

When including code snippets, make sure you use appropriate code formatting using backtics. Have a look at the markdown reference for details: https://guides.github.com/features/mastering-markdown/.

**Comment by kshitijyad** [comment]
Hi, I tried using the predict function and also used the dot product of user and item embedding, and my results were coming out different. Could you please let me know how is predict function calculating recommendations ? I was under the impression it using the dot product.

Here is my code:

Predict Function:

```
predictions=Tunedmodel.predict(user,np.arange(len(itemID_to_index)))
print(predictions)
recommended = (-predictions).argsort()[:5]
df[df.id.isin([index_to_itemID[x] for x in recommended])]
```

Using dot product:

```
customer_id='xyz'
user_vecs=Tunedmodel.user_embeddings
item_vecs=Tunedmodel.item_embeddings
customer_list=np.array(list(userID_to_index.keys()))
item_list=np.array(list(itemID_to_index.keys()))
cust_ind = np.where(customer_list == customer_id)[0][0] # Returns the index row of our customer id

rec_vector = user_vecs[cust_ind,:].dot(item_vecs.T) # Get dot product of user vector and all item vectors

recommend_vector = pref_vec*rec_vector
recommended = np.argsort(recommend_vector)[::-1][:5]
df[dfid.isin([index_to_itemID[x] for x in recommended)]

```

**Comment by maciejkula** [comment]
It looks like you might have forgotten to use the item biases. Have a look at the `get_representations` [functions](https://lyst.github.io/lightfm/docs/lightfm.html#lightfm.LightFM.get_item_representations).

**Comment by kshitijyad** [comment]
Thank you for getting back @maciejkula.
So just to be clear, do we need to multiply the item biases with item embedding and same with user embedding?

**Comment by kshitijyad** [comment]
I think I got it: 

`model = LightFM(learning_rate=0.05, loss="bpr", random_state=2019)
model.fit_partial(l.train, epochs=10)

user_biases = model.get_user_representations()[0]
user_embeddings = model.get_user_representations()[1]
item_biases = model.get_item_representations()[0]
item_embeddings = model.get_item_representations()[1]
user1 = user_embeddings[1]
item1 = item_embeddings[1]
score = user1.dot(item1.T)
ub1 = user_biases[1]
ib1 = item_biases[1]

result = score + ub1 + ib1
predict = model.predict(1, [1])

assert np.allclose(result, predict[0])`

Used this example. Thank you.
